x/auction/types/auctions.go

```diff
@@ -292,18 +314,28 @@ func (wa WeightedAddresses) Validate() error {
 	if len(wa.Weights) < 1 {
 		return fmt.Errorf("must be at least 1 weighted address")
 	}
+
 	if len(wa.Addresses) != len(wa.Weights) {
 		return fmt.Errorf("number of addresses doesn't match number of weights, %d ≠ %d", len(wa.Addresses), len(wa.Weights))
 	}
+
 	totalWeight := sdk.ZeroInt()
-	for _, w := range wa.Weights {
-		if w.IsNegative() {
-			return fmt.Errorf("weights contain a negative amount: %s", w)
+	for i := range wa.Addresses {
+		if wa.Addresses[i].Empty() {
+			return fmt.Errorf("address %d cannot be empty", i)
 		}
-		totalWeight = totalWeight.Add(w)
+		if len(wa.Addresses[i]) != sdk.AddrLen {
+			return fmt.Errorf("address %d has an invalid length: expected %d, got %d", i, sdk.AddrLen, len(wa.Addresses[i]))
+		}
+		if wa.Weights[i].IsNegative() {
+			return fmt.Errorf("weight %d contains a negative amount: %s", i, wa.Weights[i])
+		}
+		totalWeight = totalWeight.Add(wa.Weights[i])
 	}
+
 	if !totalWeight.IsPositive() {
 		return fmt.Errorf("total weight must be positive")
 	}
+
```

Comment:
Could the `WeighteAddresses.Validate` function be used here? These checks could be moved into that function

Reply:
this is the `Validate()` function